### PR TITLE
Adding WFSS Background Unit Tests

### LINKED ISSUE #3329: Background pipeline step needs unit-testing
Issue [JP-620](https://jira.stsci.edu/browse/JP-620) was created by Macarena Garcia Marin:

The background step in the JWST calibration pipeline does currently not have any test in the GitHub repository. To ensure the robustness of the pipeline and have a lower level of risk, the pipeline testing focus group is requesting unit tests to be included and routinely executed. Verifying individual code components will make easier the process of validating the pipeline. The requested code-coverage is 90%.

### PR DESCRIPTION
**Pull request by mfixstsci**
Adding NIRCAM and NIRISS WFSS mode unit tests.

Resolves #3329

### CODE REVIEW

**Comment by hbushouse** [comment]
Fixes https://jira.stsci.edu/browse/JP-620

**Comment by jdavies-st** [comment]
Fix `flake8` failures to get the CI testing to pass.

**Comment by mfixstsci** [comment]
@jdavies-st I still haven't got all of the kinks worked out quite yet. I'll add WIP to the title.

**Comment by mfixstsci** [comment]
There are tests for both `NRC/WFSS` and `NIS/WFSS` modes that are ready for review. The tests for the `NRC` are skipped because they return incorrect measurements for the background retrieved from the reference file. We are tracking that issue in [JP-1136](https://jira.stsci.edu/browse/JP-1136) but for now the `NIS` modes are passing and can be added to the suite.

**Comment by mfixstsci** [comment]
Assuming everything checks out, these should be ready for review. The problem filters were `['F480M', 'F322W2', 'F300M']` and they were removed from the list of filters to test.

File: jwst/background/tests/test_background.py [hunk 1]
```diff
@@ -134,3 +146,162 @@ def test_nirspec_gwa_ytitl(_jail, background, science_image):
...
+        'nframes': 1,
+        'ngroups': 8,
+        'nints': 1,
+        'nresets_between_ints': 0,
+        'nsamples': 1,
+        'sample_time': 10.0,
+        'start_time': 58668.72509857639,
+        'zero_frame': False}
+
+    subarray = {'xsize':2048,
```
Comment (philhodge) [inline jwst/background/tests/test_background.py]: If there's anything that actually uses the subarray information, you'll need `xstart` and `ystart` (both = 1) as well.

File: jwst/background/tests/test_background.py [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
 import pytest
+import numpy as np
+import os
 from numpy.testing.utils import assert_allclose
-from jwst.background import BackgroundStep
 
 from jwst import datamodels
+from jwst.assign_wcs import AssignWcsStep
+from jwst.background import BackgroundStep
+from jwst.background.tests import data
```
[the remark this answers is not in the document]
Comment (mfixstsci) [reply jwst/background/tests/test_background.py]: Sounds good, I will make the change

File: jwst/background/tests/test_background.py (hunk 1 above)
Comment (mfixstsci) [reply jwst/background/tests/test_background.py]: gotcha, will make the change

**Comment by mfixstsci** [comment]
@philhodge Looks like the `pr-head` check passed but I see that I have less code coverage compared to the most recent commit. Do I need to rebase?

**Comment by philhodge** [comment]
I wouldn't worry about the test coverage at this time.  Since there are no conflicts and you haven't changed anything other than this step, you shouldn't need to rebase.

**Comment by mfixstsci** [comment]
@philhodge okay cool, then these should be good to go.

**Pull request merged**